src/routes/choice-exercises/components/SingleChoiceExercise.tsx

```diff
@@ -104,30 +111,39 @@ const ChoiceExerciseScreen = ({
         navigation={navigation}
         route={route}
         currentWord={currentWord}
-        numberOfWords={documents.length}
-      />
-      {documents[currentWord]?.document_image.length > 0 && (
-        <ImageCarousel images={documents[currentWord]?.document_image} />
-      )}
-      <AudioPlayer document={documents[currentWord]} disabled={selectedAnswer === null} />
-      <SingleChoice
-        answers={answers}
-        onClick={onClickAnswer}
-        correctAnswer={correctAnswer}
-        selectedAnswer={selectedAnswer}
-        delayPassed={delayPassed}
+        numberOfWords={documents?.length ?? 0}
       />
-      <ButtonContainer>
-        {selectedAnswer !== null && (
-          <Button onPress={onFinishWord} buttonTheme={BUTTONS_THEME.dark}>
+      <Loading isLoading={loading}>
+        <>
+          <ErrorMessage error={error} refresh={refresh} />
+          {documents && correctAnswer && (
             <>
-              <LightLabelInput>
-                {currentWord + 1 >= count ? labels.exercises.showResults : labels.exercises.next}
-              </LightLabelInput>
+              {documents[currentWord]?.document_image.length > 0 && (
```

Comment:
```suggestion
              {currentDocument.document_image.length > 0 && (
```
and include `currentDocument` in check above. Same for other usages.